From: David Hildenbrand <david@redhat.com>
Subject: mm/mm_init: initialize page->_mapcount directly in __init_single_page()
Date: Wed, 29 May 2024 13:19:04 +0200

Let's simply reinitialize the page->_mapcount directly.  We can now get
rid of page_mapcount_reset().

Link: https://lkml.kernel.org/r/20240529111904.2069608-7-david@redhat.com
Signed-off-by: David Hildenbrand <david@redhat.com>
Tested-by: Sergey Senozhatsky <senozhatsky@chromium.org>	[zram/zsmalloc workloads]
Cc: Hyeonggon Yoo <42.hyeyoo@gmail.com>
Cc: Matthew Wilcox (Oracle) <willy@infradead.org>
Cc: Mike Rapoport (IBM) <rppt@kernel.org>
Cc: Minchan Kim <minchan@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/mm.h |    5 -----
 mm/mm_init.c       |    2 +-
 2 files changed, 1 insertion(+), 6 deletions(-)

--- a/include/linux/mm.h~mm-mm_init-initialize-page-_mapcount-directly-in-__init_single_page
+++ a/include/linux/mm.h
@@ -1206,11 +1206,6 @@ static inline int folio_entire_mapcount(
 	return atomic_read(&folio->_entire_mapcount) + 1;
 }
 
-static inline void page_mapcount_reset(struct page *page)
-{
-	atomic_set(&(page)->_mapcount, -1);
-}
-
 /**
  * page_mapcount() - Number of times this precise page is mapped.
  * @page: The page.
--- a/mm/mm_init.c~mm-mm_init-initialize-page-_mapcount-directly-in-__init_single_page
+++ a/mm/mm_init.c
@@ -568,7 +568,7 @@ void __meminit __init_single_page(struct
 	mm_zero_struct_page(page);
 	set_page_links(page, zone, nid, pfn);
 	init_page_count(page);
-	page_mapcount_reset(page);
+	atomic_set(&page->_mapcount, -1);
 	page_cpupid_reset_last(page);
 	page_kasan_tag_reset(page);
 
_
